Assets/TinyMCE/JoplinLists/src/main/ts/api/Commands.ts

```diff
@@ -48,6 +49,26 @@ const register = function (editor) {
   editor.addQueryStateHandler('InsertDefinitionList', queryListCommandState(editor, 'DL'));
 
   addJoplinChecklistCommands(editor, ToggleList);
+
+  const editorClickHandler = (event) => {
+    if (!isJoplinChecklistItem(event.target)) return;
+
+    // We only process the click if it's within the checkbox itself (and not the label).
+    // That checkbox, based on
+    // the current styling is in the negative margin, so offsetX is negative when clicking
+    // on the checkbox itself, and positive when clicking on the label. This is strongly
+    // dependent on how the checkbox is styled, so if the style is changed, this might need
+    // to be updated too.
+    // For the styling, see:
+    // packages/renderer/MdToHtml/rules/checkbox.ts
+    //
+    // The previous solution was to use "pointer-event: none", which mostly work, however
+    // it means that links are no longer clickable when they are within the checkbox label.
+    if (event.offsetX >= 0) return;
+
+    editor.execCommand('ToggleJoplinChecklistItem', false, { element: event.target });
+  }
+  editor.on('click', editorClickHandler);
```

Comment:
This file is to define the commands, so I don't think it should go here? Isn't there a place where various handlers or other initialisation code are setup?

Reply:
Handlers for keyboard are setup in `core/Keyboard.ts` and for buttons in `ui/Button.ts`. However this code is not related to buttons or keyboard so I don't think it should go in these files. There is no other handlers or initialization files.
What I can do is to create a file `Mouse.ts` in `core` for this handler. Is it ok for you?

Reply:
Yes a Mouse.ts file would be a good idea for this.

Reply:
I moved the changes in a Mouse.ts file.